flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkContainer.java

```diff
@@ -126,6 +133,33 @@ public void afterTest(TestDescription description, Optional<Throwable> throwable
         temporaryFolder.delete();
     }
 
+    @Override
+    protected void doStart() {
+        super.doStart();
+        try {
+            final Configuration clientConfiguration = new Configuration();
+            clientConfiguration.set(RestOptions.ADDRESS, getHost());
+            clientConfiguration.set(RestOptions.PORT, getMappedPort(JOB_MANAGER_REST_PORT));
+            this.restClusterClient =
+                    new RestClusterClient<>(clientConfiguration, StandaloneClusterId.getInstance());
+        } catch (Exception e) {
+            throw new IllegalStateException(
+                    "Failed to create client for FlinkContainer cluster", e);
+        }
```

Comment:
How about lazy creation of the client in the getter? The field can serve as a cache.